src/main/java/com/powsybl/openloadflow/ac/AcLoadFlowParameters.java

```diff
@@ -6,87 +6,123 @@
  */
 package com.powsybl.openloadflow.ac;
 
-import com.powsybl.math.matrix.MatrixFactory;
 import com.powsybl.openloadflow.OpenLoadFlowParameters;
 import com.powsybl.openloadflow.ac.equations.AcEquationSystemCreationParameters;
-import com.powsybl.openloadflow.ac.nr.NewtonRaphsonParameters;
 import com.powsybl.openloadflow.ac.outerloop.AcOuterLoop;
+import com.powsybl.openloadflow.ac.solver.AcSolverFactory;
+import com.powsybl.openloadflow.ac.solver.NewtonRaphsonFactory;
+import com.powsybl.openloadflow.ac.solver.NewtonRaphsonParameters;
 import com.powsybl.openloadflow.lf.AbstractLoadFlowParameters;
 import com.powsybl.openloadflow.network.LfNetworkParameters;
+import com.powsybl.openloadflow.network.util.UniformValueVoltageInitializer;
 import com.powsybl.openloadflow.network.util.VoltageInitializer;
 
+import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 
 /**
  * @author Geoffroy Jamgotchian {@literal <geoffroy.jamgotchian at rte-france.com>}
  */
-public class AcLoadFlowParameters extends AbstractLoadFlowParameters {
+public class AcLoadFlowParameters extends AbstractLoadFlowParameters<AcLoadFlowParameters> {
 
-    public static final int DEFAULT_MAX_OUTER_LOOP_ITERATIONS = 20;
+    private AcEquationSystemCreationParameters equationSystemCreationParameters = new AcEquationSystemCreationParameters();
 
-    private final AcEquationSystemCreationParameters equationSystemCreationParameters;
+    private NewtonRaphsonParameters newtonRaphsonParameters = new NewtonRaphsonParameters();
 
-    private final NewtonRaphsonParameters newtonRaphsonParameters;
+    private List<AcOuterLoop> outerLoops = Collections.emptyList();
 
-    private final List<AcOuterLoop> outerLoops;
+    private int maxOuterLoopIterations = DEFAULT_MAX_OUTER_LOOP_ITERATIONS;
 
-    private final int maxOuterLoopIterations;
+    private VoltageInitializer voltageInitializer = new UniformValueVoltageInitializer();
```

Comment:
Be careful here because it is also given through `LoadFlowParameters`.

Reply:
yes but with the enum so not easy to share